static/app/views/dashboardsV2/widgetBuilder/widgetBuilder.tsx

```diff
@@ -891,6 +884,19 @@ function WidgetBuilder({
     ? fields.map((field, index) => explodeField({field, alias: fieldAliases[index]}))
     : [...explodedColumns, ...explodedAggregates];
 
+  const metricFields: string[] =
+    state.dataSet === DataSet.RELEASE
+      ? state.queries.reduce((acc, query) => {
+          for (const field of [...query.aggregates, ...query.columns]) {
+            const fieldParameter = /\(([^)]*)\)/.exec(field)?.[1];
+            if (fieldParameter && !acc.includes(fieldParameter)) {
+              acc.push(fieldParameter);
+            }
+          }
+          return acc;
+        }, [] as string[])
+      : [];
```

Comment:
This looks complex, you should write a comment on what it is doing. I would prefer to also take this out of the component body and maybe just move it to the top of the file as a utility function. Tests would be highly appreciated for these sorts of functions too

Reply:
I can add a comment, but as I mentioned before, we will be re-visiting this again to make improvements

Reply:
Yeah, I would encourage you to do this. If not for others, maybe even just for yourself. Sometimes, things happen and we have to reprioritize work and you dont want to come back to this a month from now asking yourself "what did I want to do here?" (even though I'm sure that right now you know exactly what you wanted to do). There's also a chance that a teammate or a new hire may pick up the task of improving this it'll be a scary code for them to try and modify without any context

Reply:
yes I agree. I've updated created a function in the utility file and moved the code 😉